offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -898,6 +938,192 @@ export class CkbHandler {
     }
   }
 
+  // watch the ckb_unlock table and handle the new unlock events
+  // send tx according to the data
+  handleUnlockRecords(generator: CkbTxGenerator): void {
+    if (this.role !== 'collector') {
+      return;
+    }
+    const ownerTypeHash = getOwnerTypeHash();
+    this.handleTodoUnlockRecords(ownerTypeHash, generator);
+  }
+
+  handleTodoUnlockRecords(ownerTypeHash: string, generator: CkbTxGenerator): void {
+    foreverPromise(async () => this.todoUnlockRecordsHandler(ownerTypeHash, generator), {
+      onRejectedInterval: 15000,
+      onResolvedInterval: 15000,
+      onRejected: (e: Error) => {
+        logger.error(`CKB handleTodoUnlockRecords error:${e.stack}`);
+      },
+    });
+  }
+
+  async todoUnlockRecordsHandler(ownerTypeHash: string, generator: CkbTxGenerator): Promise<void> {
+    if (!TransferOutSwitch.getInstance().getStatus()) {
+      logger.info('TransferOutSwitch is off, skip handleTodoUnlockRecords');
+      return;
+    }
+    if (!this.syncedToStartTipBlockHeight()) {
+      logger.info(
+        `wait until syncing to startBlockHeight, lastHandledBlockHeight: ${this.lastHandledBlockHeight}, startTipBlockHeight: ${this.startTipBlockHeight}`,
+      );
+      return;
+    }
+    logger.debug('CkbHandler watchUnlockEvents get new unlock events and send tx');
+    const records = await this.getUnlockRecords('todo');
+    if (records.length === 0) {
+      logger.info('wait for todo unlock records');
+      await asyncSleep(3000);
+      return;
+    }
+    logger.info(`CkbHandler watchUnlockEvents unlock records: ${JSON.stringify(records)}`);
+    await this.ckbIndexer.waitForSync();
+    await this.doHandleUnlockRecords(records, generator);
+  }
+
+  async doHandleUnlockRecords(records: ICkbUnlock[], generator: CkbTxGenerator): Promise<void> {
+    if (records.length === 0) {
+      return;
+    }
+
+    const unlockTxHashes = records
+      .map((unlockRecord) => {
+        return unlockRecord.burnTxHash;
+      })
+      .join(', ');
+
+    logger.info(
+      `CkbHandler doHandleUnlockRecords start process unlock Record, burnTxHashes:${unlockTxHashes} num:${records.length}`,
+    );
+
+    records.map((r) => {
+      r.status = 'pending';
+    });
+    await this.db.saveCollectorCkbUnlock(records);
+    await this.ckbIndexer.waitForSync();
+    const allTxSkeletonWithRecords = await generator.unlock(records);
+    if (!allTxSkeletonWithRecords) {
+      return;
+    }
+    logger.debug(`unlock for records, allTxSkeletonWithRecords`, records, allTxSkeletonWithRecords);
+    logger.debug(
+      `unlock tx txSkeletons: [ ${allTxSkeletonWithRecords
+        .map((txs) => transactionSkeletonToJSON(txs.txSkeleton))
+        .join(', ')} ]`,
+    );
+    for (const txSkeletonWithRecords of allTxSkeletonWithRecords) {
+      for (;;) {
```

Comment:
Need refactor according to: https://github.com/nervosnetwork/force-bridge/blob/074c4617065236bd531508e04373b21038802a3a/offchain-modules/packages/x/src/handlers/ckb.ts#L511

Reply:
refactored, cause of unlock only one kind asset, refactored according to ckbMintHandler